From 26188686d846f392818a5396009229e18a2050b3 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Wed, 19 Aug 2020 14:51:52 +0200
Subject: [PATCH] Disable workaround MLK-11444 HYP-20171

The workaround MLK-11444 ata fixes Error ERR009598 documented in errata.
This significatly decreases SATA write speed. As long as the error is not
observed, the workaround can be deactivated.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/ata/ahci_imx.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/ata/ahci_imx.c b/drivers/ata/ahci_imx.c
index bf53b8784ab7..ebf8cdf24ed6 100644
--- a/drivers/ata/ahci_imx.c
+++ b/drivers/ata/ahci_imx.c
@@ -1574,6 +1574,12 @@ static int imx_ahci_probe(struct platform_device *pdev)
 		writel(reg_val, hpriv->mmio + IMX_TIMER1MS);
 	}
 
+	/* The following workaround fixes Error ERR009598 documented in errata.
+	 * This significatly decreases SATA write speed.
+	 * As long as the error is not observed, the workaround can be deactivated.
+	 * HYP-20171 MST
+	 */
+#if 0
 	/*
 	 * Due to IP bug on the Synopsis 3.00 SATA version,
 	 * which is present on mx6q, and not on mx53,
@@ -1591,6 +1597,7 @@ static int imx_ahci_probe(struct platform_device *pdev)
 			goto disable_sata;
 		}
 	}
+#endif
 
 	if (imxpriv->type == AHCI_IMX8QM && bist_enable) {
 		dev_info(dev, "AHCI SATA compliance test patterns.\n");
-- 
2.16.4

